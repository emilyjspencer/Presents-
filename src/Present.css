.presentform {
    width: 200px;
}
