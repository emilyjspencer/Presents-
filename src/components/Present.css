.presentform {
    display: block;
    flex-direction: row;
    font-family: Dancing Script;
    position: relative;
    left: 100px;
  
  
}

#remove-button {
    width: 10%;
    background-color: rgb(207, 188, 188);
    position: relative;
    left: 240px;
    top: -55px;
    border-radius: 5%;
    border: solid 3px rgb(165, 161, 161);
    font-family: Dancing Script;
    font-size: 16px;

}


.form-group {
    width: 90%;
  }

  #person{
    width: 30%;
    margin: auto;
    height: 25px;
    background-color: rgb(202, 187, 187);
    border-radius: 5%;
    border: solid 3px rgb(165, 161, 161);
  
}

#gift {
    width: 30%;
    margin: auto;
    height: 25px;
    background-color: rgb(202, 187, 187);
    border-radius: 5%;
    border: solid 3px rgb(163, 161, 161);
    position: relative;
    left: 8px;
}
  