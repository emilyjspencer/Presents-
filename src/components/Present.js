import React, { Component } from 'react';

import { Form, FormControl, FormGroup, FormLabel, Button } from 'react-bootstrap';

class Present extends Component {

    constructor(props) {
        super(props)
        this.state = {person: '', present: ''};
    }

    render() {
      return (
          <>
        <div className="presentform">
            
            <Form>
               <FormGroup>
                   <FormLabel>Name</FormLabel>
                   <FormControl onChange={event => this.setState({ person: event.target.value })} className="name" />
                   <br />
                   <FormLabel>Gift</FormLabel>
                   <FormControl onChange={event => this.setState({ present: event.target.value })} className="present" />
               </FormGroup>
           </Form>
           <Button 
             className="remove-button"
             onClick={() => this.props.removeItem(this.props.present.id)}
             >Remove Gift
             </Button>
           
        </div>
      
        </>
  
    )
  }
}



export default Present;