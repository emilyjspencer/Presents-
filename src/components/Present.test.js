import React from 'react';
import { shallow } from 'enzyme'
import Present from './Present'

describe('Present', () => {

    const mockRemove = jest.fn();
    const id = 1;
    const props = { present: { id }, removeItem: mockRemove };

    let wrapper;

    beforeEach(() => wrapper = shallow(<Present {...props} />));

    it('renders a <div />', () => {
        expect(wrapper.find('div').length).toEqual(1);
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('has no person associated with it at first', () => {
        expect(wrapper.state()).toEqual({ person: '', present: ''});
    });

    describe('when entering a persons name', () => {
        
        let wrapper;

        beforeEach(() => wrapper = shallow(<Present />));

        beforeEach(() => {
            wrapper.find('.name').simulate('change', { target: { value: 'Mari' }});
        });

        it('updates the name value in the component state', () => {
            expect(wrapper.state().person).toEqual('Mari');
        });
        
    }
    )

    describe('when entering a gift', () => {
        
        let wrapper;

        beforeEach(() => wrapper = shallow(<Present />));

        beforeEach(() => {
            wrapper.find('.present').simulate('change', { target: { value: 'phone'}});

            
        });

        it('updates the present in the state', () => {
            expect(wrapper.state().present).toEqual('phone');
        });
    })

    describe('clicking the remove item button', () => {

        beforeEach(() => {
            wrapper.find('.remove-button').simulate('click');
        });

        it('the removePresent callback is invoked upon button click', () => {
          expect(mockRemove).toHaveBeenCalledWith(id)
        });
    });


});