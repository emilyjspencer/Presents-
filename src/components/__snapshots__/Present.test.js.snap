// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Present should render correctly 1`] = `ShallowWrapper {}`;

exports[`Presents should render correctly 1`] = `ShallowWrapper {}`;
